actor/todo-sql/src/lib.rs

```diff
@@ -0,0 +1,261 @@
+        format!("/api/{}", id),
+        input.title,
+        input.order.unwrap_or(0),
+    );
+
+    let _resp = client
+        .execute(&ctx, &format!(r#"
+        insert into {} (id, url, title, order, completed) values ('{}','{}', '{}', {})
+        "#, TABLE_NAME, id, &todo.url, &todo.title, todo.priority))
+        .await?;
+
+    Ok(todo)
+}
+
+/// updates an existing to-do item
+async fn update_todo(ctx: &Context, client: &Db, url: &str, update: UpdateTodo) -> Result<Todo, SqlDbError> {
+    info!("Updating a todo...");
+
+    check_safety("url", url)?;
+    if let Some(title) = &update.title {
+        check_safety("title", &title)?;
+    }
+    let todo = get_todo(ctx, client, url).await?;
+    let todo = todo.update(update);
+
+    let _resp = client
+        .execute(&ctx, &format!(r#"
+        update {} set title='{}', priority={} where url='{}';
+        "#, TABLE_NAME, &todo.title, todo.priority, &todo.url))
+        .await?;
+    Ok(todo)
+}
+
+async fn get_all_todos(ctx: &Context, client: &Db) -> Result<Vec<Todo>, SqlDbError> {
+    info!("Getting all todos...");
+    let resp = client
+        .fetch(
+            &ctx,
+            &format!("select url, title, completed, order from {}", TABLE_NAME)
+        )
+        .await?;
+    let rows: Vec<Todo> = decode(&resp.rows)?;
+    Ok(rows)
+}
+
+async fn get_todo(ctx: &Context, client: &Db, url: &str) -> Result<Todo, SqlDbError> {
+    info!("Getting a todo...");
+    check_safety("url", url)?;
+    let resp = client
+        .fetch(
+            &ctx,
+            &format!("select url, title, completed, order from {} where url='{}'", TABLE_NAME, url)
+        )
+        .await?;
+    if resp.num_rows == 0 {
+        return Err(SqlDbError::new("notFound", "url not found".to_string()));
+    }
+    let mut rows: Vec<Todo> = decode(&resp.rows)?;
+    let todo = rows.remove(0);
+    Ok(todo)
+}
+
+async fn delete_all_todos(ctx: &Context, client: &Db) -> Result<(), SqlDbError> {
+    info!("Deleting all todos...");
+
+    let _resp = client
+        .execute(&ctx, &format!("delete from {}", TABLE_NAME))
+        .await?;
+
+    Ok(())
+}
+
+async fn delete_todo(ctx: &Context, client: &Db, url: &str) -> Result<(), SqlDbError> {
+    info!("Deleting a todo...");
+
+    check_safety("url", url)?;
+    let _resp = client
+        .execute(&ctx, &format!("delete from {} where url='{}'", TABLE_NAME, url))
+        .await?;
+
+    Ok(())
+}
+
+async fn handle_request(ctx: &Context, req: &HttpRequest) -> RpcResult<HttpResponse> {
+    debug!("incoming req: {:?}", req);
+
+    let trimmed_path = req.path.trim_end_matches('/');
+    let db = SqlDbSender::new();
+
+    match (req.method.as_ref(), trimmed_path) {
+        ("GET", "/") => Ok(HttpResponse {
+            body: "todo server lives at /api.  Initialize empty table with /init".to_string().into_bytes(),
+            ..Default::default()
+        }),
+        ("GET", "/init") => match create_table(ctx, &db).await {
```

Comment:
RESTy nitpick - you shouldn't use a `GET` method to produce state changes, this should be a POST imho.

Reply:
Also `create_table` should be idempotent because people can hit this URL multiple times.

Reply:
- yes on create table - will fix that and make it a POST